pkg/config/replica_config.go

```diff
@@ -279,3 +284,53 @@ func isSinkCompatibleWithSpanReplication(u *url.URL) bool {
 	return u != nil &&
 		(strings.Contains(u.Scheme, "kafka") || strings.Contains(u.Scheme, "blackhole"))
 }
+
+// AdjustEnableOldValue adjust the old value configuration by the sink scheme and encoding protocol
+func (c *ReplicaConfig) AdjustEnableOldValue(scheme, protocol string) {
+	if sink.IsMySQLCompatibleScheme(scheme) {
+		return
+	}
+
+	if c.EnableOldValue {
+		_, ok := ForceDisableOldValueProtocols[protocol]
+		if ok {
+			log.Warn("Attempting to replicate with old value enabled, but the specified protocol must disable old value. "+
+				"CDC will disable old value and continue.", zap.String("protocol", protocol))
+			c.EnableOldValue = false
+		}
+		return
+	}
+
+	_, ok := ForceEnableOldValueProtocols[protocol]
```

Comment:
why does the canal-json require disabling the old value?

Reply:
It follow the previous convention.